# Limpeza dos dados - TMDB 5000 Filmes

## Objetivo
Tratar problemas apontados nas observações da exploração

## O que será feito
- Remover colunas homepage e tagline.
- Converter valores da realese_data de str para data.
- Tratar os valores 0 e transformalos em NaN, para assim podermos prever a receita.

In [53]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [54]:
# Carregando o dataset original
df = pd.read_csv('../data/raw/tmdb_5000_movies.csv')

# cópia do dataframe para limpeza
df_limpo = df.copy()

In [55]:
# Remove as colunas homepage e tagline, pois não são úteis para a análise
df_limpo.drop(columns=['homepage', 'tagline'], inplace=True)

In [56]:
# Mostra que as colunas foram removidas do dataframe limpo
df_limpo.shape

(4803, 18)

In [57]:
# Converte a coluna release_date que estava como str para data 
df_limpo['release_date'] = pd.to_datetime(df_limpo['release_date'], errors='coerce')
# Cria a coluna release_year a partir da coluna release_date
df_limpo['release_year'] = df_limpo['release_date'].dt.year
# mostra o valor da coluna para saber se deu certo a conversão
df_limpo['release_date'].dtype

dtype('<M8[us]')

In [58]:
# Substitui zeros por ausente nas colunas financeiras
df_limpo['budget'] = df_limpo['budget'].replace(0, np.nan)
df_limpo['revenue'] = df_limpo['revenue'].replace(0, np.nan)

# Remove linhas sem dados financeiros válidos
df_limpo = df_limpo.dropna(subset=['budget', 'revenue'])

df_limpo.shape

(3229, 19)

In [59]:
# Salva o dataset limpo para uso nos próximos notebooks
df_limpo.to_csv('../data/processed/tmdb_limpo.csv', index=False)

## Observações Finais

- Criado um df_limpo na pasta processed, para não mexer no original
- Colunas homepage e tagline removidas
- Dados da coluna release_data convertido de str para data
- Nova coluna chamada release_year
- Dados com valores 0, foram substituidos por ausentes (NaN)
- Dados com valor NaN, foram removidos, deixando no total 3229 linha e 19 colunas
- Próximo notebook vamos fazer a ánalise do projeto com gráficos